bounties_api/std_bounties/tests.py

```diff
@@ -281,6 +387,71 @@ def test_activate_bounty(self):
             pk=self.bounty_to_activate_id)
         self.assertEqual(result, activated_bounty_from_db)
 
+    def test_fulfill_bounty(self):
+        bounty_to_fulfill = Bounty.objects.get(pk=self.bounty_to_fulfill_id)
+        fulfillment_timestamp = '1517536922'
+        fulfillment_datetime = datetime.fromtimestamp(int(fulfillment_timestamp))
+        fulfillment_id = 10
+        issuer = '0x4242424242424242424242424242424242424242'
+        inputs = {
+            'data': 'QmQjchBM6tjAvXzkDEpWgLUv9Ui4jwqtxsEzB6LxB2WqFL'
+        }
+        fulfillment = self.client.fulfill_bounty(
+            bounty=bounty_to_fulfill,
+            fulfillment_id=fulfillment_id,
+            inputs=inputs,
+            event_timestamp=fulfillment_timestamp,
+            transaction_issuer=issuer)
+
+        self.assertEqual(fulfillment.fulfillment_id, fulfillment_id)
+        self.assertEqual(fulfillment.bounty.id, self.bounty_to_fulfill_id)
+        self.assertEqual(fulfillment.accepted, False)
+        self.assertEqual(fulfillment.fulfiller, issuer)
+        self.assertEqual(fulfillment.fulfillment_created, fulfillment_datetime)
+
+        # Try fulfill with duplicate id
+        fulfillment = self.client.fulfill_bounty(
+            bounty=bounty_to_fulfill,
+            fulfillment_id=fulfillment_id,
+            inputs=inputs,
+            event_timestamp=fulfillment_timestamp,
+            transaction_issuer=issuer)
+        self.assertIsNone(fulfillment)
+
+    def test_accept_fulfillment(self):
+        bounty_to_fulfill = Bounty.objects.get(pk=self.bounty_to_fulfill_id)
+        fulfillment_timestamp = '1517536922'
+        fulfillment_datetime = datetime.fromtimestamp(int(fulfillment_timestamp))
+        fulfillment_to_accept = Fulfillment.objects.get(pk=self.fulfillment_to_accept_id)
+        fulfillment = self.client.accept_fulfillment(
+            bounty=bounty_to_fulfill,
+            fulfillment_id=fulfillment_to_accept.fulfillment_id,
+            event_timestamp=fulfillment_timestamp)
+        self.assertEqual(fulfillment.bounty.bountyStage, COMPLETED_STAGE)
+        self.assertEqual(fulfillment.accepted, True)
+        self.assertEqual(fulfillment.accepted_date, fulfillment_datetime)
+        self.assertEqual(bounty_to_fulfill.balance, 0)
+
+    def test_kill_bounty(self):
+        bounty_to_kill = Bounty.objects.get(pk=self.bounty_to_kill_id)
+        fulfillment_timestamp = '1517536922'
+        bounty = self.client.kill_bounty(
+            bounty=bounty_to_kill,
+            event_timestamp=fulfillment_timestamp)
+        self.assertEqual(bounty.bountyStage, COMPLETED_STAGE)
+
+    def test_add_contribution(self):
+        bounty_to_top_up = Bounty.objects.get(pk=self.bounty_to_top_up_id)
+        inputs = {
+            'value': 1
+        }
+        fulfillment_timestamp = '1517536922'
+        bounty = self.client.add_contribution(
```

Comment:
What do you think of also testing these scenarios? 

- Adding a contribution to a bounty that has run out of funds by a fulfillment being accepted
- Adding a contribution that still does not add up to the fulfillment amount, so it should not become active after all

For example a bounty that has `0.99` in funds, which is not enough for the `1` fulfillment amount